client/my-sites/email/email-non-owner-message/index.tsx

```diff
@@ -0,0 +1,98 @@
+import { SiteDetails } from '@automattic/data-stores';
+import { TranslateResult, useTranslate } from 'i18n-calypso';
+import { stringify } from 'qs';
+import { useEmailOwnerUserName } from 'calypso/components/data/query-email-owner-username';
+import TrackComponentView from 'calypso/lib/analytics/track-component-view';
+import { recordTracksEvent } from 'calypso/lib/analytics/tracks';
+import { emailManagementEdit } from 'calypso/my-sites/email/paths';
+
+import './style.scss';
+
+type EmailNonOwnerMessageProps = {
+	domainName: string;
+	selectedSite?: SiteDetails | null;
+};
+
+const buildQueryString = ( parameters = {} ) =>
+	parameters ? stringify( parameters, { addQueryPrefix: true, skipNulls: true } ) : '';
+
+export const EmailNonOwnerMessage = ( props: EmailNonOwnerMessageProps ) => {
+	const { domainName, selectedSite } = props;
+
+	const translate = useTranslate();
+
+	const ownerUserName = useEmailOwnerUserName( selectedSite, domainName );
+
+	const buildLoginUrl = () => {
+		const redirectUrlParameter = emailManagementEdit(
+			selectedSite?.slug ?? '',
+			domainName,
+			'manage',
+			null,
+			'login-redirect'
+		);
+
+		return `/log-in/${ buildQueryString( {
+			email_address: ownerUserName,
+			redirect_to: redirectUrlParameter,
+		} ) }`;
+	};
+
+	const loginUrl = buildLoginUrl();
+
+	const onClickLink = ( eventType: 'login' | 'support' ) => {
+		const properties = {
+			action: eventType,
+		};
+
+		recordTracksEvent( `calypso_email_providers_nonowner_click`, properties );
+	};
+
+	const translateOptions = {
+		components: {
+			contactSupportLink: (
+				<a
+					href="https://wordpress.com/help/contact"
+					onClick={ () => onClickLink( 'support' ) }
+					rel="noopener noreferrer"
+					target="_blank"
+				/>
+			),
+			loginLink: <a href={ loginUrl } onClick={ () => onClickLink( 'login' ) } rel="external" />,
+			strong: <strong />,
+		},
+		args: {
+			ownerUserName,
+			selectedDomainName: domainName,
+		},
+	};
+
+	let reasonText: TranslateResult | null;
+
+	if ( ownerUserName ) {
+		reasonText = translate(
+			'Additional mailboxes can only be purchased by {{strong}}%(ownerUserName)s{{/strong}}, ' +
+				'who is the owner of the email subscription. ' +
+				'If you have access to that account, please {{loginLink}}log in with the account{{/loginLink}} to make a purchase. ' +
+				'Otherwise, please reach out to {{strong}}%(ownerUserName)s{{/strong}} or {{contactSupportLink}}contact support{{/contactSupportLink}}.',
+			translateOptions
+		);
+	} else {
+		reasonText = translate(
+			'Additional mailboxes can only be purchased by the owner of the email subscription. ' +
+				'If you have access to that account, please log in with the account to make a purchase. ' +
+				'Otherwise, please {{contactSupportLink}}contact support{{/contactSupportLink}}.',
+			translateOptions
+		);
+	}
+
+	return (
+		<>
+			<TrackComponentView
+				eventName="calypso_non_email_owner_notice_component_impression"
```

Comment:
```suggestion
				eventName="calypso_email_providers_nonowner_impression"
```
Same comments from the main PR regarding tracking (https://github.com/Automattic/wp-calypso/pull/68158#discussion_r984688935)

Reply:
Let's rename this event.